codes/javascript/chapter_graph/graph_adjacency_matrix.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * File: graph_adjacency_matrix.js
+ * Created Time: 2023-02-09
+ * Author: Zhuo Qinyue (1403450829@qq.com)
+ */
+
+/* 基于邻接矩阵实现的无向图类 */
+class GraphAdjMat {
+    vertices;     // 顶点列表，元素代表“顶点值”，索引代表“顶点索引”
+    adjMat; // 邻接矩阵，行列索引对应“顶点索引”
+
+    /* 构造函数 */
+    constructor(vertices, edges) {
+        this.vertices = [];
+        this.adjMat = [];
+        // 添加顶点
+        for (const val of vertices) {
+            this.addVertex(val);
+        }
+        // 添加边
+        // 请注意，edges 元素代表顶点索引，即对应 vertices 元素索引
+        for (const e of edges) {
+            this.addEdge(e[0], e[1]);
+        }
+    }
+
+    /* 获取顶点数量 */
+    size() {
+        return this.vertices.length;
+    }
+
+    /* 添加顶点 */
+    addVertex(val) {
+        const n = this.size();
+        // 向顶点列表中添加新顶点的值
+        this.vertices.push(val);
+        // 在邻接矩阵中添加一行
+        const newRow = [];
+        for (let j = 0; j < n; j++) {
+            newRow.push(0);
+        }
+        this.adjMat.push(newRow);
+        // 在邻接矩阵中添加一列
+        for (const row of this.adjMat) {
+            row.push(0);
+        }
+    }
+
+    /* 删除顶点 */
+    removeVertex(index) {
+        if (index >= this.size()) {
+            return;
```

Comment:
```suggestion
            throw new RangeError("Index Out Of Bounds Exception");
```

Reply:
TypeScript has the same problem. Please address them, thank you!

Reply:
A question.
why not use `return;` but use `Error`
`return;` isn't equivalent to `void`?
@krahets @justin-tse

Reply:
We have to let the user know the operation can not be executed.

if using `return`. It is better to make the function return None and return None when the index is out of range.

Reply:
OK. I got it.
Thanks for your answer! @krahets

Reply:
If the function is only used to throw the error, then we will use type: never.
Otherwise, we will use the remaining type. Because anything | never is anything. So we will use void here.

Reply:
thanks @justin-tse